### PR TITLE
TechDebt: Misc Odd Updates (keycloak, pipeline, favicon, etc)

### PR DESCRIPTION
**Pull request by NickPhura**
# Overview

## Links to Jira tickets

n/a

## Description of relevant changes

- Removed some old git workflow files
- Updated keycloak version, which changed it to always being defined (rather than potentially undefined), and so I removed a bunch of unnecessary `?.` when dereferencing the keycloak object.
- Removed a bunch of null user object checks, as the user object would never be undefined.
- Added `change_version` to the version endpoint (mainly an experimental change, as I never really noticed that var before, but in theory it might let us put a version number in the UI)
- Pulled some taxonomy-service changes from other projects, just misc updated
- Added the BCGov favicon to the react app.
- Added proper error handling to the db setup pipeline step, so it will properly fail the git action if it fails in openshift
- Moved a bunch of old tests from manually specifying mock keycloak auth states, to using the `getMockAuthState` test helper function

## PR Checklist

A list of items that are good to consider when making any changes.

_Note: this list is not exhaustive, and not all items are always applicable._

### Code

- [ ] New files/classes/functions have appropriately descriptive names and comment blocks to describe their use/behaviour
- [ ] I have avoided duplicating code when possible, moving re-usable pieces into functions
- [ ] I have avoided hard-coding values where possible and moved any re-usable constants to a constants file
- [ ] My code is as flat as possible (avoids deeply nested if/else blocks, promise chains, etc)
- [ ] My code changes account for null/undefined values and handle errors appropriately
- [ ] My code uses types/interfaces to help describe values/parameters/etc, help ensure type safety, and improve readability

### Style

- [ ] My code follows the established style conventions
- [ ] My code uses native material-ui components/icons/conventions when possible

### Documentation

- [ ] I have commented my code sufficiently, such that an unfamiliar developer could understand my code
- [ ] I have added/updated README's and related documentation, as needed

### Tests

- [ ] I have added/updated unit tests for any code I've added/updated
- [ ] I have added/updated the Postman requests/tests to account for any API endpoints I've added/updated

### Linting/Formatting

- [ ] I have run the linter and fixed any issues, as needed  
       _See the `lint` commands in package.json_
- [ ] I have run the formatter and fixed any issues, as needed  
       _See the `format` commands in package.json_

### SonarCloud

- [ ] I have addressed all SonarCloud Bugs, Vulnerabilities, Security Hotspots, and Code Smells

### CODE REVIEW

File: app/src/hooks/useKeycloakWrapper.tsx [hunk 1]
```diff
@@ -47,10 +47,10 @@ export interface IKeycloakWrapper {
   /**
    * Original raw keycloak object.
    *
-   * @type {(Keycloak | undefined)}
+   * @type {(Keycloak)}
    * @memberof IKeycloakWrapper
    */
-  keycloak: Keycloak | undefined;
+  keycloak: Keycloak;
```
Comment (curtisupshall) [inline app/src/hooks/useKeycloakWrapper.tsx:54]: +1 for no more undefined keycloak 🙏

**Review by curtisupshall — DISMISSED** [review]
Code looks great! Lots of welcome changes

**Review by curtisupshall — DISMISSED** [review]

**Review by AlfredRosenthal — DISMISSED** [review]
Ran it locally, tried all the areas that were touched, it looks good

🦺

**Comment by NickPhura** [comment]
@curtisupshall @AlfredRosenthal Thanks for the reviews, I had to fix some conflicts so will need a re-review. Nothing functional should have changed. I updated some tests while I was at it to use `getMockAuthState` rather than hardcoding it in each test. Will see how that evolves long term, but it seemed like we were having issues maintaining our mock auth states and so hopefully this will help centralize it a bit.

**Review by curtisupshall — APPROVED** [review]

**Review by al-rosenthal — APPROVED** [review]
🕶️

**Pull request merged**